machine/v1/types_nutanixprovider.go

```diff
@@ -154,6 +165,139 @@ type NutanixResourceIdentifier struct {
 	Name *string `json:"name,omitempty"`
 }
 
+// NutanixGPUIdentifierType is an enumeration of different resource identifier types for GPU entities.
+// +kubebuilder:validation:Enum:=Name;DeviceID
+type NutanixGPUIdentifierType string
+
+const (
+	// NutanixGPUIdentifierName identifies a GPU by Name.
+	NutanixGPUIdentifierName NutanixGPUIdentifierType = "Name"
+
+	// NutanixGPUIdentifierDeviceID identifies a GPU by device ID.
+	NutanixGPUIdentifierDeviceID NutanixGPUIdentifierType = "DeviceID"
+)
+
+// NutanixGPU holds the identity of a Nutanix GPU resource in the Prism Central
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'DeviceID' ?  has(self.deviceID) : !has(self.deviceID)",message="deviceID configuration is required when type is DeviceID, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Name' ?  has(self.name) : !has(self.name)",message="name configuration is required when type is Name, and forbidden otherwise"
+// +union
+type NutanixGPU struct {
+	// type is the identifier type to use for this GPU resource.
+	// The GPU device can be identified either by name or deviceID.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Type NutanixGPUIdentifierType `json:"type"`
+
+	// deviceID is the GPU device ID with the integer value.
+	// +optional
+	// +unionMember
+	DeviceID *int32 `json:"deviceID,omitempty"`
+
+	// name is the GPU device name
+	// +optional
+	// +unionMember
+	Name *string `json:"name,omitempty"`
+}
+
+// NutanixDiskFlashMode is an enumeration of different disk flash modes.
+// +kubebuilder:validation:Enum=On;Off
+type NutanixDiskFlashMode string
+
+const (
+	// NutanixDiskFlashModeOn represents the disk flash mode is On.
+	NutanixDiskFlashModeOn NutanixDiskFlashMode = "On"
+
+	// NutanixDiskFlashModeOff represents the disk flash mode is Off.
+	NutanixDiskFlashModeOff NutanixDiskFlashMode = "Off"
+)
+
+// NutanixVMStorageConfig specifies the storage configuration parameters for VM disks.
+type NutanixVMStorageConfig struct {
+	// flashMode specifies whether to pin the VM disk to the flash tier.
+	// +kubebuilder:default=Off
+	FlashMode NutanixDiskFlashMode `json:"flashMode"`
+
+	// storageContainer refers to the storage_container used by the VM disk.
+	// Currently the storageContainer configuration is supported only by using its UUID.
+	// In the future, we may support to identify storageContainer by its name.
```

Comment:
I wouldn't include this, but instead add a different UUID only resource identifier struct that only allows the UUID for now

Reply:
Changed to use a new struct NutanixStorageResourceIdentifier allowing UUID for now, but expandable in the future.